packages/super-components/components/msc-billing-tile/src/msc-billing-tile/msc-billing-contact.tsx

```diff
@@ -0,0 +1,197 @@
+import {
+  Component,
+  h,
+  Prop,
+  State,
+  Host,
+  Element,
+  Fragment,
+} from '@stencil/core';
+import {
+  OdsThemeColorIntent,
+  OdsThemeTypographyLevel,
+  OdsThemeTypographySize,
+} from '@ovhcloud/ods-theming';
+import {
+  OdsHTMLAnchorElementTarget,
+  OdsIconName,
+  OdsIconSize,
+} from '@ovhcloud/ods-core';
+import { HTMLStencilElement } from '@stencil/core/internal';
+import { apiClient } from '@ovh-ux/manager-core-api';
+import {
+  ContactInfos,
+  DomainProperties,
+  ServiceInfos,
+  Translations,
+} from './msc-billing.types';
+
+export interface IMscBillingContact {
+  serviceName: string;
+  serviceType: string;
+  serviceInfos: ServiceInfos;
+  changeOwnerDataTracking?: string;
+  updateOwnerDataTracking?: string;
+  subscriptionManagementDataTracking?: string;
+  localeStrings: Translations;
+}
+
+@Component({
+  tag: 'msc-billing-contact',
+  styleUrl: 'msc-billing-contact.scss',
+  shadow: true,
+})
+export class MscBillingContact implements IMscBillingContact {
+  @Element() host!: HTMLStencilElement;
+
+  @Prop() serviceName: string;
+
+  @Prop() serviceType: string;
+
+  @Prop() serviceInfos: ServiceInfos;
+
+  @Prop() changeOwnerDataTracking?: string;
+
+  @Prop() updateOwnerDataTracking?: string;
+
+  @Prop() subscriptionManagementDataTracking?: string;
+
+  @Prop() localeStrings: Translations;
+
+  @State() domainProperties?: DomainProperties;
+
+  @State() ownerInfos?: ContactInfos;
+
+  async componentWillLoad() {
+    if (this.isDomainServiceType()) {
+      this.domainProperties = await apiClient.v6
+        .get(`/domain/${this.serviceName}`)
+        .then((response) => response.data);
+
+      this.ownerInfos = await apiClient.v6
+        .get(`/domain/contact/${this.domainProperties?.whoisOwner}`)
+        .then((response) => response.data);
```

Comment:
**nitpick:** The implementation is very specific to `domain` product. It causes lot of rework when we try to integrate with other products.

Reply:
I don't understand what do you suggest here ?

Reply:
`billing-super-component` is used in lot of products like Dedicated Server, VPS, Nutanix, Domain and so on.
So the APIs used here is specific to Domain which makes super-component not usable in other products.

Reply:
Why not ? The API call is conditioned according to the service type detected in the url passed as a parameter so the call won't be made on other products.

How does it make it not usable on these products ?